Application/Security.js

```diff
@@ -100,18 +100,117 @@ angular.module('shared')
           }
         );
       },
+      oAuth2Authorize: function(client_id, redirect_uri, response_type, scope, user, callback) {
+        var self = this;
+        
+        switch (response_type) {
```

Comment:
This switch statement only operates on one case statement.  I think it should be rewritten like so

```
if (!response_type) {
  return;
}

<case statement implementation here>
```

This would reduce the level of nesting by 2, and would make this a lot easier to read.